internal/common/monitor_where_creator.go

```diff
@@ -0,0 +1,41 @@
+package common
+
+import (
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/jinzhu/gorm"
+)
+
+type MonitorWhereCreator struct {
+	lastInvokeTime time.Time
+	calls          int64
+}
+
+func NewMonitorWhereCreator() *MonitorWhereCreator {
+	return &MonitorWhereCreator{}
+}
+
+func (m *MonitorWhereCreator) createWhereCondition(db *gorm.DB, diff time.Duration) *gorm.DB {
+	timeToCompare := strfmt.DateTime(time.Now().Add(-diff))
+	return db.Where("id in (select distinct(id) from (select id from clusters where updated_at > ? union select cluster_id as id from hosts where updated_at > ?) as t)", timeToCompare, timeToCompare)
+}
+
+func (m *MonitorWhereCreator) Where(db *gorm.DB) *gorm.DB {
```

Comment:
i think this is bad name, this function is doing something very specific and such a general name doesn't explain it

Reply:
Any better one?